app/components/common/Address.tsx

```diff
@@ -44,6 +46,25 @@ export function Address({
     const address = pubkey.toBase58();
     const { cluster } = useCluster();
     const addressPath = useClusterPath({ pathname: `/address/${address}` });
+    const [showNicknameEditor, setShowNicknameEditor] = useState(false);
+    const [nickname, setNickname] = useState<string | null>(null);
+
+    // Load nickname from localStorage
+    useEffect(() => {
+        setNickname(getNickname(address));
+
+        // Listen for nickname updates
+        const handleNicknameUpdate = (event: CustomEvent) => {
+            if (event.detail.address === address) {
+                setNickname(getNickname(address));
+            }
+        };
+
+        window.addEventListener('nicknameUpdated', handleNicknameUpdate as EventListener);
+        return () => {
+            window.removeEventListener('nicknameUpdated', handleNicknameUpdate as EventListener);
+        };
+    }, [address]);
```

Comment:
Could you please extract this hook into a separate file? `app/entities/address/model/use-nickname.ts` is a good place imo.